core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateActionListener.java

```diff
@@ -0,0 +1,11 @@
+package org.elasticsearch.cluster.action.shard;
+
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.transport.TransportException;
+
+public interface ShardStateActionListener {
```

Comment:
can we fold this into ShardStateAction and just call it listener there? I think we should reduce the number of satellite classes we have.